libs/causality/src/lib/CausalAnalysisDashboard/Controls/CausalAnalysisView/CausalIndividualView/CausalIndividualChart.tsx

```diff
@@ -458,6 +458,40 @@ export class CausalIndividualChart extends React.PureComponent<
         trace.y = rawY;
       }
     }
+    if (
+      this.context.jointDataset.datasetMetaData?.featureMetaData !== undefined
+    ) {
+      const identityFeatureName =
+        this.context.jointDataset.datasetMetaData.featureMetaData
+          ?.identity_feature_name;
+
+      if (identityFeatureName !== undefined) {
+        const jointDatasetFeatureName =
+          this.context.jointDataset.getJointDatasetFeatureName(
+            identityFeatureName
+          );
+
+        if (jointDatasetFeatureName !== undefined) {
+          const metaIdentityFeature =
+            this.context.jointDataset.metaDict[jointDatasetFeatureName];
+          const rawIdentityFeature = JointDataset.unwrap(
+            dictionary,
+            jointDatasetFeatureName
+          );
+          hovertemplate += `${localization.Common.identityFeature}(${metaIdentityFeature.label}): {point.customdata.ID}<br>`;
```

Comment:
will _metaIdentityFeature_ always be available? or should we add '?' in metaIdentityFeature?.label

Reply:
At this point, it should be always available to us since the variable `jointDatasetFeatureName` is returned to be valid.